lib/event.rb

```diff
@@ -0,0 +1,57 @@
+class Event
+
+  attr_reader :name,
+              :crafts,
+              :attendees
+
+  def initialize(name, crafts, attendees)
+    @name = name
+    @crafts = crafts
+    @attendees = attendees
+  end
+
+  def attendee_names
+    @attendees.map {|attendee| attendee.name}
+  end
+
+  def craft_with_most_supplies
+    craft_with_most_supplies = ""
+    craft_supplies_length = 0
+    @crafts.each do |craft|
+      if craft.supplies_required.length > craft_supplies_length
+        craft_supplies_length = craft.supplies_required.length
+        craft_with_most_supplies = craft.name
+      end
+    end
+    craft_with_most_supplies
+  end
+
+  def supply_list
+    supply_list = []
+    @crafts.each do |craft|
+      craft.supplies_required.keys.each do |key|
+        supply_list << key.to_s
+      end
+    end
+    supply_list.uniq
+  end
+
+  def crafts_that_use(supply)
+    crafts_that_use_supply = []
+
+    @crafts.each do |craft|
+      string_keys = []
+
+      craft.supplies_required.each_key do |k|
+        string_keys << k.to_s
+      end
+
+      if string_keys.include?(supply)
+        crafts_that_use_supply << craft
+      end
+    end
+    crafts_that_use_supply
+  end
+end
```

Comment:
Instead of transforming each key in `supplies_required` we can change `supply` to a symbol and check that is is included in the `supplies_required`

```suggestion
  def crafts_that_use(supply)
    crafts_that_use_supply = []

    @crafts.each do |craft|
      if craft.supplies_required.keys.include?(supply.to_sym)
        crafts_that_use_supply << craft
      end
    end
    crafts_that_use_supply
  end
end
```

even better if we use `select` to return the crafts that met our boolean criteria rather than having to create the array ourselves

```suggestion
  def crafts_that_use(supply)
    @crafts.select do |craft|
      craft.supplies_required.keys.include?(supply.to_sym)
    end
  end
end
```